pochivm/api_base.h

```diff
@@ -115,113 +115,189 @@ Value<T>::operator Value<U>() const
     return StaticCast<U>(*this);
 }
 
+// Language utility: construct a literal
+// Example: Literal<int64_t>(1)
+//
+template<typename T>
+Value<T> Literal(T x)
+{
+    return Value<T>(new AstLiteralExpr(TypeId::Get<T>(), &x));
+}
+
+// If src is a literal expression, return a new equivalent literal expression of type U. Otherwise, return
+// a StaticCast of src to U
+//
+template<typename U, typename T, typename = std::enable_if_t<AstTypeHelper::may_static_cast<T, U>::value> >
+Value<U> StaticCastOrConvertLiteral(const Value<T>& src)
+{
+    static_assert(AstTypeHelper::may_static_cast<T, U>::value, "cannot static_cast T to U");
+    if(src.__pochivm_value_ptr->GetAstNodeType() == AstNodeType::AstLiteralExpr)
+    {
+        return Literal<U>(static_cast<U>(assert_cast<AstLiteralExpr *>(src.__pochivm_value_ptr)->template GetAs<T>()));
```

Comment:
nit: remove the space between in `AstLiteralExpr *` to be consistent with the existing code style